src/scripting/scriptingService.ts

```diff
@@ -11,26 +11,281 @@ import {
     IScriptOptions,
     ScriptingProgressNotification,
     ScriptOperation,
+    ScriptingCompleteNotification,
+    ScriptingCancelRequest,
 } from "../models/contracts/scripting/scriptingRequest";
 import { TreeNodeInfo } from "../objectExplorer/nodes/treeNodeInfo";
 import * as vscode from "vscode";
 import { IScriptingObject, IServerInfo } from "vscode-mssql";
+import SqlDocumentService, { ConnectionStrategy } from "../controllers/sqlDocumentService";
+import { ObjectExplorerUtils } from "../objectExplorer/objectExplorerUtils";
+import { ActivityStatus, TelemetryActions, TelemetryViews } from "../sharedInterfaces/telemetry";
+import { startActivity } from "../telemetry/telemetry";
+import * as LocalizedConstants from "../constants/locConstants";
+import * as Constants from "../constants/constants";
+import { getUriKey } from "../utils/utils";
+import { Deferred } from "../protocol";
+import { SqlOutputContentProvider } from "../models/sqlOutputContentProvider";
+import { IConnectionProfile } from "../models/interfaces";
+import StatusView from "../views/statusView";
+import { Logger } from "../models/logger";
+import VscodeWrapper from "../controllers/vscodeWrapper";
+import { UserSurvey } from "../nps/userSurvey";
+
+export const SCRIPT_OPERATION_CANCELED_ERROR = "Scripting operation cancelled by user.";
 
 export class ScriptingService {
     private _client: SqlToolsServiceClient;
+    private _onGoingScriptingOperations: Map<
+        string,
+        Deferred<{
+            script: string;
+            errorMessage: string;
+            errorDetails: string;
+        }>
+    > = new Map();
+    private _logger: Logger;
 
-    constructor(private _connectionManager: ConnectionManager) {
+    constructor(
+        private _context: vscode.ExtensionContext,
+        private _vscodeWrapper: VscodeWrapper,
+        private _connectionManager: ConnectionManager,
+        private _sqlDocumentService: SqlDocumentService,
+        private _sqlOutputContentProvider: SqlOutputContentProvider,
+        private _statusview: StatusView,
+        private _objectExplorerTree: vscode.TreeView<TreeNodeInfo>,
+    ) {
         this._client = this._connectionManager.client;
+
+        this._logger = Logger.create(this._vscodeWrapper.outputChannel, "ObjectExplorerService");
+
+        const pushDisposable = (disposable: vscode.Disposable): void => {
```

Comment:
Can you shift these registrations and onNotifications into an `initialize()` method to make this more readable?